src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,391 @@
+
+        private bool _temperatureCompensation = false;
+
+        /// <summary>
+        /// Use temperature compensation
+        /// </summary>
+        public bool TemperatureCompensation
+        {
+            get
+            {
+                return _temperatureCompensation;
+            }
+
+            set
+            {
+                _temperatureCompensation = value;
+
+                byte retval = ReadByte((byte)Register.ISE_CONFIG_REGISTER);
+
+                retval = (byte)Bit_set((int)retval, IseTemperatureCompensationConfigBit, _temperatureCompensation);
+
+                WriteByte((byte)Register.ISE_CONFIG_REGISTER, retval);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UFireIse"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to be used</param>
+        public UFireIse(I2cDevice i2cDevice)
+        {
+            _device = i2cDevice;
+        }
+
+        /// <summary>
+        /// Measure a value from the ISE Probe Interface, typical measure are in the millivolt range.
+        /// </summary>
+        /// <returns>value from ISE Probe Interface, typical measure are in the millivolt range. On error it return -1 as value</returns>
+        public ElectricPotential Measure()
+        {
+            SendCommand((byte)Command.ISE_MEASURE_MV);
+
+            DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+
+            _measurement = ReadRegister((byte)Register.ISE_MV_REGISTER);
+            _measurement = Convert.ToSingle(Math.Round(_measurement, 2));
+
+            return new ElectricPotential(_measurement, UnitsNet.Units.ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Measure temperature
+        /// </summary>
+        /// <returns>Temperature. Temperature range: -2 to 35 C. Temperature Precision: 0.05 C</returns>
+        public Temperature MeasureTemperature()
+        {
+            SendCommand((byte)Command.ISE_MEASURE_TEMP);
+
+            DelayHelper.DelayMilliseconds(IseTemperatureMeasureTime, allowThreadYield: true);
+            _temperature = new Temperature(ReadRegister((byte)Register.ISE_TEMP_REGISTER), UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+
+            return _temperature;
+        }
+
+        /// <summary>
+        /// Set temperature compensation
+        /// </summary>
+        /// <param name="temp">Compensation temperature</param>
+        public void SetTemperatureCompensation(Temperature temp)
+        {
+            WriteRegister((byte)Register.ISE_TEMP_REGISTER, Convert.ToSingle(temp.DegreesCelsius));
+            _temperature = temp;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single point using a mV value.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateSingle(ElectricPotential solution)
```

Comment:
this needs more explanation. Calibrate to what? are you marking where 0 is? If this needs longer explanation please make sure at least sample explains how to use it